### PR TITLE
Add / edit mileage page

### PR DESCRIPTION
**Pull request by mvaishnavi**

### CODE REVIEW

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 1]
```diff
@@ -697,8 +690,8 @@ export class AddEditMileagePage implements OnInit {
 
     const defaultMileage$ = forkJoin({
       defaultVehicle: defaultVehicle$,
-      orgSettings: this.offlineService.getOrgSettings(),
-    }).pipe(map(({ defaultVehicle, orgSettings }) => orgSettings.mileage[defaultVehicle]));
+      mileageRates: this.offlineService.getMileageRates(),
```
Comment (mvaishnavi) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:693]: some clean up around not referencing `this.offlineService.getMileageRates()` everywhere but using the allowed mileage rate everywhere is done here - https://github.com/fylein/fyle-mobile-app/pull/1268/files

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 2]
```diff
@@ -966,7 +973,20 @@ export class AddEditMileagePage implements OnInit {
...
-    this.mileageRates$ = this.getMileageRates();
+    this.mileageRates$ = this.offlineService.getMileageRates();
+
+    this.mileageRatesOptions$ = forkJoin({
+      mileageRates: this.mileageRates$,
+      homeCurrency: this.homeCurrency$,
+    }).pipe(
+      map(({ mileageRates, homeCurrency }) =>
+        mileageRates.map((rate) => {
+          const unit = rate.unit && rate.unit.toLowerCase() === 'miles' ? 'mile' : 'km';
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:984]: This block of code is repeated -lets extract it to a function and move it to a service

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 3]
```diff
@@ -845,6 +838,20 @@ export class AddEditMileagePage implements OnInit {
     }
   }
 
+  formatMileageRateName(rateName) {
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:841]: move to service

**Review by Julias0 — CHANGES_REQUESTED** [review]
minor comments

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 4]
```diff
@@ -633,25 +638,16 @@ export class AddEditMileagePage implements OnInit {
...
-    const options = [];
-    if (mileageConfig.two_wheeler) {
-      options.push('two_wheeler');
-    }
-
-    if (mileageConfig.four_wheeler) {
-      options.push('four_wheeler');
-    }
+  getRateByVehicleType(mileageRates, vehicle_type) {
+    const filteredMileageRate = mileageRates.filter((mileageRate) => mileageRate.vehicle_type === vehicle_type);
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:642]: You can use the `.find` to do this

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 5]
```diff
@@ -633,25 +638,16 @@ export class AddEditMileagePage implements OnInit {
...
-      options.push('four_wheeler1');
-    }
+    return filteredMileageRate && filteredMileageRate.length > 0 && filteredMileageRate[0].rate;
+  }
 
-    if (mileageConfig.four_wheeler4) {
-      options.push('four_wheeler4');
-    }
+  getMileageByVehicleType(mileageRates, vehicle_type) {
+    const filteredMileageRate = mileageRates.filter((mileageRate) => mileageRate.vehicle_type === vehicle_type);
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:648]: same as above

**Review by Julias0 — APPROVED** [review]
Minor comment - fix and merge

**Comment by mvaishnavi** [comment]
will merge to master along with - https://github.com/fylein/fyle-mobile-app/pull/1268

**Pull request closed without merging**